src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -759,10 +772,27 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
             ᶜq = MatrixFields.get_field(ᶜspecific, q_name)
             ∂ᶜρq_err_∂ᶜρ = matrix[ρq_name, @name(c.ρ)]
             ∂ᶜρq_err_∂ᶜρq = matrix[ρq_name, ρq_name]
+            ᶜtridiagonal_matrix_scalar = ifelse(
+                q_name in (:q_rai, :q_sno),
+                ᶜdiffusion_h_matrix_scaled,
+                ᶜdiffusion_h_matrix,
+            )
             @. ∂ᶜρq_err_∂ᶜρ =
-                dtγ * ᶜdiffusion_h_matrix ⋅ DiagonalMatrixRow(-(ᶜq) / ᶜρ)
+                dtγ * ᶜtridiagonal_matrix_scalar ⋅ DiagonalMatrixRow(-(ᶜq) / ᶜρ)
             @. ∂ᶜρq_err_∂ᶜρq +=
-                dtγ * ᶜdiffusion_h_matrix ⋅ DiagonalMatrixRow(1 / ᶜρ)
+                dtγ * ᶜtridiagonal_matrix_scalar ⋅ DiagonalMatrixRow(1 / ᶜρ)
```

Comment:
@trontrytel Both of these two versions work (without an increase in allocation). Compared to the commented out version, this one has fewer lines, but needs one more scratch variable. Which one do you prefer?

Reply:
I don't have a strong preference. I would vote for the currently uncommented one.